File: examples/companion-packages/scala-maven-app/src/main/scala/ai/agentxm/examples/pawmatch/Flags.scala
package ai.agentxm.examples.pawmatch

import ai.agentxm.examples.tinyflags.{Flags, booleanFlag, createFlags, variantFlag}

/** Flag keys used by the PawMatch CLI. */
object FlagKeys:
  val HomeCheckFollowup = "home-check-followup"
  val FeeBreakdownDetailed = "fee-breakdown-detailed"
  val LongStayHighlight = "long-stay-highlight"
  val SuggestDonateAfterAdoption = "suggest-donate-after-adoption"
  val ShowCharityRatings = "show-charity-ratings"
  val RecommendationStrategy = "recommendation-strategy"
  val MatchQuizDepth = "match-quiz-depth"
  val PetCardStyle = "pet-card-style"
  val DonateFocusDefault = "donate-focus-default"

/** Canonical PawMatch [[Flags]] bundle. */
def createPawMatchFlags(): Flags = createFlags(
  FlagKeys.HomeCheckFollowup -> booleanFlag(default = false, rollout = Some(25)),
  FlagKeys.FeeBreakdownDetailed -> booleanFlag(default = true),
  FlagKeys.LongStayHighlight -> booleanFlag(default = true),
  FlagKeys.SuggestDonateAfterAdoption -> booleanFlag(default = false, rollout = Some(50)),
  FlagKeys.ShowCharityRatings -> booleanFlag(default = true),
  FlagKeys.RecommendationStrategy -> variantFlag(
    variants = List("popularity", "match-quiz", "longest-stay"),
    default = Some("match-quiz"),
    rollout = Some(Map("longest-stay" -> 20)),
  ),
  FlagKeys.MatchQuizDepth -> variantFlag(
    variants = List("short", "standard", "thorough"),
    default = Some("standard"),
  ),
  FlagKeys.PetCardStyle -> variantFlag(
    variants = List("compact", "detailed", "playful"),
    default = Some("detailed"),
  ),
  FlagKeys.DonateFocusDefault -> variantFlag(
    variants = List("all", "shelters", "rescue"),
    default = Some("all"),
  ),
)

File: examples/companion-packages/maven-kotlin-app/src/main/kotlin/ai/agentxm/examples/pawmatch/Flags.kt
package ai.agentxm.examples.pawmatch

import ai.agentxm.examples.tinyflags.Flags
import ai.agentxm.examples.tinyflags.booleanFlag
import ai.agentxm.examples.tinyflags.createFlags
import ai.agentxm.examples.tinyflags.variantFlag

object FlagKeys {
    const val HOME_CHECK_FOLLOWUP = "home-check-followup"
    const val FEE_BREAKDOWN_DETAILED = "fee-breakdown-detailed"
    const val LONG_STAY_HIGHLIGHT = "long-stay-highlight"
    const val SUGGEST_DONATE_AFTER_ADOPTION = "suggest-donate-after-adoption"
    const val SHOW_CHARITY_RATINGS = "show-charity-ratings"
    const val RECOMMENDATION_STRATEGY = "recommendation-strategy"
    const val MATCH_QUIZ_DEPTH = "match-quiz-depth"
    const val PET_CARD_STYLE = "pet-card-style"
    const val DONATE_FOCUS_DEFAULT = "donate-focus-default"
}

fun createPawMatchFlags(): Flags = createFlags(
    FlagKeys.HOME_CHECK_FOLLOWUP to booleanFlag(default = false, rollout = 25),
    FlagKeys.FEE_BREAKDOWN_DETAILED to booleanFlag(default = true),
    FlagKeys.LONG_STAY_HIGHLIGHT to booleanFlag(default = true),
    FlagKeys.SUGGEST_DONATE_AFTER_ADOPTION to booleanFlag(default = false, rollout = 50),
    FlagKeys.SHOW_CHARITY_RATINGS to booleanFlag(default = true),
    FlagKeys.RECOMMENDATION_STRATEGY to variantFlag(
        variants = listOf("popularity", "match-quiz", "longest-stay"),
        default = "match-quiz",
        rollout = mapOf("longest-stay" to 20),
    ),
    FlagKeys.MATCH_QUIZ_DEPTH to variantFlag(
        variants = listOf("short", "standard", "thorough"),
        default = "standard",
    ),
    FlagKeys.PET_CARD_STYLE to variantFlag(
        variants = listOf("compact", "detailed", "playful"),
        default = "detailed",
    ),
    FlagKeys.DONATE_FOCUS_DEFAULT to variantFlag(
        variants = listOf("all", "shelters", "rescue"),
        default = "all",
    ),
)
